src/observations/jpl_eph.jl

```diff
@@ -124,20 +124,28 @@ const ttmtdb = TaylorInterpolant(sseph.t0, sseph.t, sseph.x[:,end])
 
 @doc raw"""
     loadpeeph(et::Union{Nothing, Real} = nothing)
+    loadpeeph(et_0::Real, et_f::Real)
 
-Load Solar System ephemeris produced by `PlanetaryEphemeris.jl` from Jan 1st 2000 up to `et > 0` [ephemeris seconds since J2000].
+Load Solar System ephemeris produced by `PlanetaryEphemeris.jl` in timerange `[0, et]` (`[et_0, et_f]`) where `et` must have units
+of ephemeris seconds since J2000.
 
-**Caution**: running this function for the first time will download the `sseph_p100` artifact (∼554 MB) which can take several minutes.
+**Caution**: running this function for the first time will download the `sseph_p100` artifact (∼556 MB) which can take several minutes.
 """
 function loadpeeph(et::Union{Nothing, Real} = nothing)
     if isnothing(et)
         return sseph
     else
-        idxs = findall(x -> x <= et, sseph.t)
-        return TaylorInterpolant(sseph.t0, sseph.t[idxs], sseph.x[idxs[1:end-1], :])
+        i = searchsortedfirst(sseph.t, et)
+        return TaylorInterpolant(sseph.t0, sseph.t[1:i], sseph.x[1:i-1, :])
     end
 end
```

Comment:
Two comments (about this function):
- I think it makes sense to exploit dispatch here, and define the functions
    ```julia
    loadpeeph() = sseph
    loadpeeph(::Nothing) = loadpeeph()
    ```
    In this way, you avoid the the `Union`, the `if` and thus all branching.
- What is `sseph` here and in my previous suggestion? My point is that it is either a `global` quantity, which we should perhaps avoid, or simply undefined...

Reply:
> I think it makes sense to exploit dispatch here, and define the functions

+ 1 to the suggestion of using multiple dispatch 👍

Reply:
Regarding `sseph`, I suggest to type-annotate it, to help the compiler with type-inference:
```julia
const sseph::TaylorInterpolant{Float64,Float64,2} = JLD2.load(sseph_artifact_path, "ss16ast_eph")
```
We could even consider adding type annotations to `loadpeeph`:
```julia
loadpeeph()::TaylorInterpolant{Float64,Float64,2} = sseph
loadpeeph(::Nothing)::TaylorInterpolant{Float64,Float64,2} = loadpeeph()
```
The rationale behind using `sseph` is to load the the Solar System ephemeris  via artifacts only once in memory.